pkgs/top-level/haskell-ng.nix

```diff
@@ -16,7 +17,10 @@ rec {
     ghc722 = callPackage ../development/compilers/ghc/7.2.2.nix { ghc = compiler.ghc704Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
     ghc742 = callPackage ../development/compilers/ghc/7.4.2.nix { ghc = compiler.ghc704Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
     ghc763 = callPackage ../development/compilers/ghc/7.6.3.nix { ghc = compiler.ghc704Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
-    ghc784 = callPackage ../development/compilers/ghc/7.8.4.nix { ghc = compiler.ghc742Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
+    ghc784 = callPackage ../development/compilers/ghc/7.8.4.nix {
+      ghc = if pkgs.stdenv.isDarwin then compiler.ghc783Binary else compiler.ghc742Binary;
```

Comment:
The older compiler doesn't work properly when clang provides `cpp`. I'd be happy to revise the patch so 783Binary is not actually exposed, but only used to compile 784 on darwin.